File: Chapter 5/CSharpCarLibrary/HotRod.cs
using System;
using System.Windows.Forms;

namespace CSharpCarLibrary
{
	/// <summary>
	/// Summary description for HotRod.
	/// </summary>
	public class HotRod : Car, IConvertible
	{
		// IConvertible impl.
		// (using explicit interface impl).
		void IConvertible.OpenSunRoof(bool openIt)
		{
			if(openIt)
				MessageBox.Show("Sun roof is open!", "C# Car Library");
			else
				MessageBox.Show("Closing sun roof...", "C# Car Library");
		}
		
		#region Overrides
		// Overrides.
		public override void SpeedUp()
		{
			base.mCurrSpeed += 20;
		}
		public override void DisplayBumperStickerText()
		{
			MessageBox.Show("Taking names and kickin' butt...", "C# Car Library");
		}		
		#endregion 

		#region Constructors
		public HotRod()
		{}
		public HotRod(string name, CarColor color) 
			: base(name, color)
		{}
		public HotRod(string name, CarColor color, int sp) 
			: base(name, color, sp)
		{}
		#endregion 
	}
}

File: Chapter 5/VbNetCarLibrary/HotRod.vb
Public Class HotRod
    Inherits Car
    Implements IConvertible

    ' IConvertible impl.
    Public Overridable Sub OpenSunRoof(ByVal openIt As Boolean) Implements IConvertible.OpenSunRoof
        If (openIt) Then
            MessageBox.Show("Sun roof is open!", "VB.NET Car Library")
        Else
            MessageBox.Show("Closing sun roof...", "VB.NET Car Library")
        End If
    End Sub

#Region "Constructors"
    Public Sub New()
    End Sub

    Public Sub New(ByVal name As String, ByVal color As CarColor)
        MyBase.new(name, color)
    End Sub
    Public Sub New(ByVal name As String, ByVal color As CarColor, ByVal sp As Integer)
        MyBase.new(name, color, sp)
    End Sub
#End Region

#Region "Overrides"
    ' Overrides.
    Public Overrides Sub SpeedUp()
        mCurrSpeed += 20
    End Sub
    Public Overrides Sub DisplayBumperStickerText()
        MessageBox.Show("Taking names and kickin' butt...", "VB.NET Car Library")
    End Sub
#End Region

End Class